### PR TITLE
833 refactor the create account form

### LINKED ISSUE #833: [REFACTORING] Refactor the createAccount form
In order to follow the principle of least astonishment, providing refactoring of createAccount form component is required.
All hardcoded validators should be removed. Instead, utilize ```packages/venia-concept/src/util/formValidators.js``` as a single approach for validation. 
Basic implementation has been done in magento-research/pwa-studio#770.

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
## Description
PR contains refactoring of createAccount component

## Related Issue
Closes #833  [REFACTORING] Refactor the createAccount form 

## Motivation and Context
In order to follow the principle of least astonishment, I've provided refactoring of createAccount form component, removed hardcoded validation, updated tests.

## Screenshots (if appropriate):
image

## Proposed Labels for Change Type/Package
REFACTORING
venia-concept

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [x] I have linked an issue to this PR.
- [x] I have indicated the change type and relevant package(s).
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All CI checks are green (linting, build/deploy, etc).
- [ ] At least one core contributor has approved this PR.

### CODE REVIEW

**Comment by tjwiebell** [comment]
@supernova-at - This appears to overlap enough with #913 that one of us would have a terrible conflict resolution ahead of us. Going to mark as hold, and link to #913.

**Comment by supernova-at** [comment]
Good call :thumbsup:. 

I tried to hold off changing much about this page in #913 knowing that this PR was coming.

I'll take ownership of it since I was just working in this code.

File: packages/venia-concept/src/components/CreateAccount/asyncValidators.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        const available = await request('/rest/V1/customers/isEmailAvailable', {
+            method: 'POST',
+            body: JSON.stringify(body)
+        });
+
+        return !available ? 'This email address is not available.' : null;
+    } catch (error) {
+        throw 'An error occurred while looking up this email address.';
+    }
+};
```
Comment (supernova-at) [inline packages/venia-concept/src/components/CreateAccount/asyncValidators.js]: The next version of `informed` that we're moving to in #1078 does not support `asyncValidators`. We have decided to keep this code around in the hopes that support for it gets added back. Then we can just hook up this code to the UI.

**Comment by supernova-at** [comment]
@VitaliyBoyko thanks so much for this! Sorry for the delay; since it has been so long since we looked at this I went ahead and pushed a couple minor commits instead of asking you to change anything.

We'll get this reviewed and merged ASAP 😃

**Comment by tjwiebell** [comment]
@supernova-at - Please resolve conflicts with latest mainline and look into the unit test failure reported by Danger.

**Pull request merged**